### DISCUSSION TITLE
사장일 때 지원현황 페이지

### DISCUSSION THREAD

**Discussion by gayeong0916** [team-final/albaform#133] [General]
![화면 캡처 2024-10-01 012559](https://github.com/user-attachments/assets/d93a4fe1-325e-4df9-8197-19a02ef5e6e9)

사장일 때 파란색 부분이 안 보이는데, 저 부분을 '회원의 내 지원 내역 조회'라 본인만 조회가 가능한 부분인데
그러면 사장은 어느 api를 가지고 와서 해야하는걸까요...?
사장일 땐 '지원 상세 조회' api를 통해서 가지고 오면 되는 거 같은데 그럼 지원자 상세 페이지에다가 applicantId를 전달해줘야 하는데...
뭔가 정확하게 이해가 안 가서 내일 얘기하면서 이 궁금증을 풀고 싶어욥...!

제출 내용 / 지원일시 & 진행 상태 각각 이 부분들을 사장일 때와 지원자일 때 두 번을 생각해서 api를 연동하는 것이 맞는것인지...

**Comment by JuhyeokC** [comment]
첨부하신 이미지는
1. 사장님이
2. 본인이 생성한 알바폼 페이지에서
3. 지원 현황블록의 지원자 중
4. 지원자를 클릭했을 때
이동하는 페이지로 보입니다.

>GET `/{teamId}/forms/{formId}/applications` 지원 현황 목록 조회
```json
{
  "nextCursor": 0,
  "data": [
    {
      "applicantId": 0,
      "updatedAt": "2024-09-30T18:12:25.342Z",
      "createdAt": "2024-09-30T18:12:25.342Z",
      "status": "REJECTED",
      "introduction": "string",
      "resumeName": "string",
      "resumeId": 0,
      "experienceMonths": 0,
      "phoneNumber": "string",
      "name": "string",
      "id": 0
    }
  ]
}
```

지원 현황 목록 조회 API 에서 얻은 `applicantId` 로 아래 API 로 GET 요청을 하면될 것 같아요.

>GET `/{teamId}/applications/{applicationId}` 지원 상세 조회
```json
{
  "applicantId": 0,
  "updatedAt": "2024-09-30T18:18:13.361Z",
  "createdAt": "2024-09-30T18:18:13.361Z",
  "status": "REJECTED",
  "introduction": "string",
  "resumeName": "string",
  "resumeId": 0,
  "experienceMonths": 0,
  "phoneNumber": "string",
  "name": "string",
  "id": 0
}
```

위 데이터에서 `createdAt` 이 지원일시 `status` 가 진행상태로 보입니다.
![image](https://github.com/user-attachments/assets/6a0b0b2e-c492-4469-937b-e63826b1008c)
`status` 값으로는 `[ REJECTED, INTERVIEW_PENDING, INTERVIEW_COMPLETED, HIRED ]` 이렇게 네가지 데이터로 분류되네요
![image](https://github.com/user-attachments/assets/165db165-d3c9-4266-b67f-817a213adf13)
순서대로 `[거절, 면접 대기, 면접 완료, 채용 완료]` 로 변환할 수 있는 것 같습니다.
>객체로 생성해두면 좋겠네요
```ts
export const FORM_STATUS = {
  REJECTED: '거절',
  INTERVIEW_PENDING: '면접 대기',
  INTERVIEW_COMPLETED: '면접 완료',
  HIRED: '채용 완료',
} as const

export type FormStatusType = keyof typeof FORM_STATUS
export type FormStatusWord = (typeof FORM_STATUS)[FormStatusType]
```

정리하면.. 사장일 때 '지원 현황 목록 조회' API 로 받은 데이터에서 `applicantId` 값으로 '지원 상세 조회' API 를 요청하면 말씀하신 데이터를 받아올 수 있는 것 같습니다.

같은 페이지이지만 지원자, 사장님 두가지 사용자 유형에 맞춰 API 를 요청하는게 맞는 것 같습니다.

**Reply by gayeong0916** [reply]
오호!! 알겠습니댜!!
